pepper-apis/src/main/java/org/broadinstitute/ddp/util/ElasticsearchServiceUtil.java

```diff
@@ -71,6 +72,7 @@ public static synchronized RestHighLevelClient getElasticsearchClient(Config cfg
                     httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
                     if (proxyUrl != null) {
                         httpClientBuilder.setProxy(new HttpHost(proxyUrl.getHost(), proxyUrl.getPort(), proxyUrl.getProtocol()));
+                        httpClientBuilder.setConnectionReuseStrategy(NoConnectionReuseStrategy.INSTANCE);
```

Comment:
This change is based on [this](https://stackoverflow.com/questions/52997697/how-to-get-around-connection-reset-by-peer-when-using-elasticsearchs-restclie) and [this](https://stackoverflow.com/questions/39755456/how-to-configure-squid-connectionreadwrite-timeout-per-request-instead-of-glob).

My theory is that the HTTP client here is keeping connections around when on the other side the Squid proxy had already closed the connection. I suspect we're not seeing the `connection reset` issue on every single request is because the HTTP client cycles through a pool of connections, and we just so happens to miss out on the closed connection sometimes. From those 2 SO questions, it sounds like for HTTPS traffic, Squid is not able to inspect the keep-alive headers and so will automatically close connections. Meanwhile, the Apache HTTP client has some fancy built-in logic for doing keep-alive, and when there's no keep-alive header (as is the case for Elasticsearch) it will keep connections around indefinitely. I don't know what the keep-alive time is for these various systems, so I opted to not reuse connections.

Previously, we're creating and closing the ES client on every single request. So moving to "no connection reuse" will effectively give us the same behavior as before. However, we still get some benefits since there's underlying resources like thread pools so we don't create those every single time.